Automated release with Semantic Release (#32)
inpsyde/reusable-workflows: docs/release-semantic-automated-npm.md

> # Release semantic automated
> 
> This workflow utilizes the [Semantic Release](https://github.com/semantic-release/semantic-release) package to create a release of a package that uses npm.
> Mind that, in order to use Semantic Release, you need to stick to their convention about commit messages.
> 
> You can provide a release.config.js file in your repository to create a custom release that uses the following Semantic Release plugins:
> - git
> - npm
> - exec
> 
> Otherwise, the workflow will create the release with a standard set of configurations, updating the version in the following files:
> - style.css or index.php 
> - CHANGELOG.md
> - package-lock.json
> - package.json
> - composer.json
> 
> If you don't ship any of these files the release process will not fail.
> 
> Another standard set of configurations is on the branches: every push to `main` and `next` branches will release a stable version, every push to `alpha` and `beta` branches will release a pre-release version.
> If you would like another setting, please provide your custom release.config.js file.
> 
> Packages based on yarn will not be released using this automation.

Reviewer: I slightly updated the wording. Any chance you can apply the max characters by line, just like the other Markdown files? It should be 100 or 120. You can use PhpStorm for this. Thanks!

```suggestion
# Automatic release

This workflow utilizes [semantic-release](https://github.com/semantic-release/semantic-release) to create a package release.
Note that you must stick to their [commit message convention](https://github.com/semantic-release/semantic-release#commit-message-format) to use it.

You can provide a `release.config.js` file in your repository to create a custom release that uses the following semantic-release plugins:
- git
- npm
- exec

Otherwise, the workflow will create the release with a standard set of configurations, updating the version in the following files:
- `CHANGELOG.md`
- `composer.json`
- `package-lock.json`
- `package.json`
- `style.css` or `index.php`

By default, every push to the `main` and `next` branches will release a stable version, and every push to the `alpha` and `beta` branches will create a pre-release version.
If you would like another setting, please provide your custom release.config.js file.

Packages based on yarn will not be released using this automation.
```

**Simplest possible example:**

```yml
name: Release
on:
  push:
    branches:
      - main
      - beta
      - alpha
jobs:
  release:
    uses: inpsyde/reusable-workflows/.github/workflows/release-semantic-automated-npm.yml@main
```

### Configuration parameters

#### Inputs

| Name                  | Default                                                                                       | Description                                          |
|-----------------------|-----------------------------------------------------------------------------------------------|------------------------------------------------------|

> | `NODE_VERSION`        | 18                                                                                            | Node version with which the release will be executed |

Reviewer: Could you please fix the table? If you reformat the file for the line length with PhpStorm (see previous comment), this should be fixed automatically.

Author: Done
